### ISSUE TITLE
Syntax highlight bug in PowerShell - not detecting string quoute close

### ISSUE THREAD

**Issue by aeschli** [PowerShell/EditorSyntax#67]
_From @csharmath on February 16, 2018 16:0_

### Issue Type
Bug

### Description
<img width="663" alt="stringsyntaxhighlightbug" src="https://user-images.githubusercontent.com/1450460/36316670-e15354d0-1308-11e8-9b60-ff57bbdba525.PNG">

Paste this to a ps1 file.

```
Get-Disk `
    | Where-Object {$_.Size -gt $MinSize -and $_.BusType -notmatch 'fibre|iscsi|usb' } `
    | Sort-Object -Property Number
# This is not a string but will be displayed as string
```

Expected:  String displayed on between the two single quoutes.
Actual: Everything is displayed as string after the opening single quoute

### VS Code Info

VS Code version: Code - Insiders 1.21.0-insider (70e020b2377f12894ca11254548ce365a0e435f1, 2018-02-15T05:17:01.988Z)
OS version: Windows_NT x64 10.0.15063

<details><summary>Extensions (16)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-markdownlint|Dav|0.13.0
xml|Dot|1.9.2
tslint|eg2|1.0.27
puppet-vscode|jpo|0.9.0
mssql|ms-|1.3.0
python|ms-|2018.1.0
csharp|ms-|1.14.0
PowerShell|ms-|1.5.1
vscode-docker|Pet|0.0.24
smartpaste|reb|0.0.3
java|red|0.19.0
vscode-icons|rob|7.19.0
vscode-java-debug|vsc|0.6.0
vscode-java-pack|vsc|0.3.0
vscode-java-test|vsc|0.3.0
vscode-maven|vsc|0.5.1

</details>
Reproduces only with extensions

_Copied from original issue: Microsoft/vscode#43839_

**Comment by csharmath** [comment]
Just noticed that there was an update for insiders, updated and it still reproduces.
Also attaching a screenshot of working VSCode vs broken VSCode insiders.
Both with PowerShell 1.5.1 extension

![image](https://user-images.githubusercontent.com/1450460/36318277-53e303e8-130d-11e8-8d49-1b0915bd8937.png)

**Comment by aeschli** [comment]
I updated the Powershell to the latest from this repo. That means we are back to what we had in 1.20 and this bug no longer present.

There are still some improvements that could be done:

![image](https://user-images.githubusercontent.com/6461412/36721283-be5b2458-1baa-11e8-93bc-38d5986c62fa.png)

**Comment by omniomi** [comment]
This is fixed as of 529562f

![fixed](https://user-images.githubusercontent.com/12937335/40325295-676cab0a-5cf8-11e8-8534-cbb84f239b75.png)
